### ISSUE TITLE
Question of the implementation of Dash

### ISSUE THREAD

**Issue by MemoriesJ** [microsoft/Semi-supervised-learning#87]
Hi,

When I was comparing the formula used in the Dash paper with your implementation,
I found that lines 49 and 50 of utils.py of the Dash is different from
i.e., 
<img width="610" alt="image" src="https://user-images.githubusercontent.com/40484277/225602935-1fa2c44c-61b0-4823-9845-db0ff41dbfb8.png">

equation 13 in the paper, 
i.e., 

<img width="374" alt="image" src="https://user-images.githubusercontent.com/40484277/225603105-aa2c705d-09b4-4659-828e-d09535481edb.png">

where you compute the loss for masking through the logits of weak augmented inputs, however, in the paper, it is computed via the logits of strong augmented view.

Please correct me if I am wrong.

Best

**Comment by Hhhhhhao** [comment]
We tried the strong augmentation at first but it doesn't work, then we switched to weak augmentation. Since back then they haven't published the code yet (but now seems they have). 

You can try strong augmentation or align with their code, if you find any performance improvement we can open a PR for this.
